src/TodosContext.tsx

```diff
@@ -0,0 +1,152 @@
+import React, { useState } from 'react';
+
+export type Todo = {
+  title: string,
+  id: number,
+  completed: boolean,
+};
+
+type Props = {
+  children: React.ReactNode,
+};
+
+export const todos: Todo[] = [];
+
+interface Context {
+  newTodos: Todo[],
+  newTodo: Todo,
+  handleTitleChange: (event: React.ChangeEvent<HTMLInputElement>) => void,
+  handleOnSubmit: (event: React.FormEvent) => void,
+  handleCompletedChange: (id: number) => void,
+  handleTodoDelete: (id: number) => void,
+  handleAllCompletedChange: (
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => void,
+  allCompleted: boolean,
+  activeTodos: Todo[],
+  handleClearCompleted: () => void,
+  completedTodos: Todo[],
+}
+
+export const TodosContext = React.createContext<Context>({
+  newTodos: [],
+  newTodo: {
+    title: '',
+    id: 0,
+    completed: false,
+  },
+  handleTitleChange: () => { },
+  handleOnSubmit: () => { },
+  handleCompletedChange: () => { },
+  handleTodoDelete: () => { },
+  handleAllCompletedChange: () => {},
+  allCompleted: false,
+  activeTodos: [],
+  handleClearCompleted: () => {},
+  completedTodos: [],
+});
+
+export const TodosProvider: React.FC<Props> = ({ children }) => {
+  const [newTodo, setNewTodo] = useState<Todo>({
+    title: '',
+    id: 0,
+    completed: false,
+  });
+
+  const [allCompleted, setAllCompleted] = useState(false);
```

Comment:
This stae is redundant. Instead of state you may use const value, same as `activeTodos` variable